### ISSUE TITLE
[Top bar] Suggestion/feedback for more flexibility

### ISSUE THREAD

**Issue by brettsmason** [zurb/foundation-sites#7907]
I've been trying to use the top bar in a WordPress theme I've just released (more to follow on that!) but am having problems using it. I wanted to see if it would be considered. If so I'll put together a PR.

Basically I would love to dscuss changing the top bar markup to:

```
<div class="top-bar">
	<div class="top-bar-branding"><!-- rather than having menu-text within a menu -->
		Logo or text here
	</div>
	<div id="menu-id-to-toggle" class="top-bar-menu"><!-- wrap the navigation -->
		<ul class="top-bar-left menu"> // or keep as divs like at the moment and nest the menu
			<ul class="menu">
			</ul>
		</ul>
		<ul class="top-bar-right menu"> // or keep as divs like at the moment and nest the menu
		</ul>
	</div>
</div>
```

Styling wise:

```
.top-bar-branding {
	float: left; // so that its displayed on the left of the top-bar-menu
}

.top-bar-menu {
	display: block; // displayed to the right of .top-bar-branding
}
```

The benefit of this is that the top-bar-menu can be toggled with the responsive toggle, but keep the branding intact. Also I find the markup a bit awkward at the moment when trying to use a logo (which I expect is fairly common) within the menu and thought this would be cleaner.

I'd love to hear thoughts on this - I can put up a Codepen if needed.

**Comment by brettsmason** [comment]
Here's an idea of what I meant: [CodePen](http://codepen.io/brettsmason/pen/WrXoOe)
Just wondered if this was something that would be considered or altered or if its more a place for custom code on my part?

**Comment by gakimball** [comment]
This is interesting, and like you said fits better with the responsive toggler plugin. We'll talk about it internally a bit.

**Comment by zurbrandon** [comment]
This is actually really cool! I'd love a pull request with the top-bar-branding class you made if you have a chance.

**Comment by brettsmason** [comment]
No problem I'll create a PR this evening. Before I do this though any preference on behaviour with the .stacked-for-xx code? This is what I thought:

```
/// makes sections stacked
@mixin top-bar-stacked {
  // Sub-sections
  .top-bar-branding {
    width: 100%;
  }

  .top-bar-right {
    width: 100%;
  }

  .top-bar-left {
    width: 100%;
  }
}

@mixin foundation-top-bar {
  // Top bar container
  .top-bar {
    @include top-bar-container;
  }

  // Generate classes for stacking on each screen size (defined in $breakpoint-classes)
  @each $size in $breakpoint-classes {
    .stacked-for-#{$size} {
      @include breakpoint($size down) {
        @include top-bar-stacked;
      }
    }
  }

  // Stack on small screens as default
  @include breakpoint(small only) {
    @include top-bar-stacked;
  }

  // Sub-sections
  .top-bar-branding {
    float: left;
    padding-right: $grid-column-gutter;
  }

  .top-bar-left {
    float: left;
  }

  .top-bar-right {
    float: right;
  }
}
```

If you could take a look or advise on anything you want to change I'll sort something out!

**Comment by andycochran** [comment]
I wonder if there's a class more semantic than `.top-bar-branding`. The context for a menu might not have a "brand." Maybe something like `.top-bar-header`? 

Relatedly, Top Bar can be used in locations other than "top" (see its placement in the docs). The name is an unfortunate holdover from v5. Maybe it should be renamed Menu Bar (`.menu-bar`, `.menu-bar-header`)?

**Comment by gakimball** [comment]
@andycochran I could see `.top-bar-title`, since that bit is probably always going to be a title area of some kind.

**Comment by brettsmason** [comment]
@andycochran I agree a more semantic name for top bar would be good, and also agree that `.menu-bar` would be nice. I guess this would be difficult though to suddenly change the name of a component?

@gakimball I agree, `-title` would be a better fit. What about the other css (padding) - is that all OK?

If everyone agrees I'll put a PR together (unless theres any discussion around name change?).

**Comment by andycochran** [comment]
Foundation has renamed things before. We'd just need to… 
```scss
@mixin foundation-top-bar {
  @include foundation-menu-bar;
  ...
}
````
…and include a deprecation notice until 6.2 (or whenever the old name is dropped).

**Comment by gakimball** [comment]
@andycochran We're trying to avoid renaming whole components within a major version. We did a *lot* of that from 5 to 6 as is, and we haven't even published a migration guide yet ;)

We also already have a few necessary migrations in place for 6.2 (responsive gutters, flexbox stuff, Sass variables), so we'd prefer to not add too much more to this version.

**Comment by gakimball** [comment]
@brettsmason The CSS you have is fine. If we need to do any fine-tuning in our testing after we pull it in we'll be able to.

**Comment by brettsmason** [comment]
@gakimball There you go :smile: 
Would there be an arguement to remove the `menu-text` class from menu? I can't see me ever using it, as I would always favour this (it always felt a bit wrong to me anyway to add the title to the menu like that).

**Comment by andycochran** [comment]
@gakimball That aversion to renaming totally makes sense. Besides, since y'all made v6 so amazing, I can simply use semantic classes if I don't like Top Bar's name.

**Comment by gakimball** [comment]
Merged #7944
